provider/factory.go

```diff
@@ -0,0 +1,26 @@
+package provider
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-go/tfprotov5"
+	"github.com/hashicorp/terraform-plugin-mux/tf5muxserver"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// ProtoV5ProviderServerFactory returns a muxed terraform-plugin-go protocol v5 provider factory function.
+// The primary (Plugin SDK) provider server is also returned (useful for testing).
+func ProtoV5ProviderServerFactory(ctx context.Context, version string) (func() tfprotov5.ProviderServer, *schema.Provider, error) {
+	primary := New(version)
+
+	servers := []func() tfprotov5.ProviderServer{
+		primary.GRPCProvider,
+	}
+
+	muxServer, err := tf5muxserver.NewMuxServer(ctx, servers...)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return muxServer.ProviderServer, primary, nil
```

Comment:
Similarly, returning the `primary` provider makes sense right now, but would start to become more confusing once the migration was actually underway. 

Happy to ignore this one for now, and remove the `acctest.Provider` variable in another PR.

Reply:
That's true!  Let me make another PR to remove the `acctest.Provider`!